core/src/main/java/com/google/errorprone/bugpatterns/LinkedListChecker.java

```diff
@@ -0,0 +1,48 @@
+package com.google.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+import static com.google.errorprone.matchers.Description.NO_MATCH;
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.Category;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker.InstanceOfTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(name = "LinkedListChecker", summary = "java.util.LinkedList accessed through via index",
+    severity = ERROR, category = Category.JDK)
+public final class LinkedListChecker extends BugChecker
+    implements MethodInvocationTreeMatcher, InstanceOfTreeMatcher {
+
+  private static final String LINKED_LIST_FQN = "java.util.LinkedList";
+  private static final Matcher<MethodInvocationTree> INSTANCE_LINKEDLIST_MATCHER = Matchers.anyOf(
+      MethodMatchers.instanceMethod().onDescendantOf(LINKED_LIST_FQN).named("add")
```

Comment:
This isn't necessarily bad. `myLinkedList.add(0, item)` is actually probably a good intended use of a `LinkedList`.

Maybe at the least special-case `0`, or even any small contant (or perhaps all constants)?

The same applies for the other methods below.